tests/IntervalTest.php

```diff
@@ -95,6 +106,135 @@ public function testGetDuration(): void
     }
 
+    /** @dataProvider providerContains */
+    public function testContains(int $start, int $end, int $now, bool $expected, string $errorMessage): void
+    {
+        $interval = new Interval(Instant::of($start), Instant::of($end));
+
+        $this->assertEquals($expected, $interval->contains(Instant::of($now)), $errorMessage);
+    }
+
+    public function providerContains(): array
+    {
+        return [
+            'at the start' => [
+                1000000000,
+                2000000000,
+                1000000000,
+                true,
+                'an Interval must contain its start',
+            ],
+            'at the end' => [
+                1000000000,
+                2000000000,
+                2000000000,
+                false,
+                'an Interval must not contain its end',
+            ],
+            'in the middle' => [
+                1000000001,
+                1000000003,
+                1000000002,
+                true,
+                'an Interval must contain its intermediate values',
+            ],
+        ];
+    }
+
+    /** @dataProvider providerIntersectsWith */
+    public function testIntersectsWith(int $start1, int $end1, int $start2, int $end2, bool $expected): void
+    {
+        $interval1 = new Interval(Instant::of($start1), Instant::of($end1));
+        $interval2 = new Interval(Instant::of($start2), Instant::of($end2));
+        $this->assertEquals($expected, $interval1->intersectsWith($interval2));
+    }
+
+    public function providerIntersectsWith(): array
+    {
+        return [
+            'second is after first' => [
+                100000, 200000,
+                400000, 500000,
+                false,
+            ],
+            'second is before first' => [
+                400000, 500000,
+                100000, 200000,
+                false,
+            ],
+            'end of the first is start of the second' => [
+                100000, 200000,
+                200000, 300000,
+                false,
+            ],
+            'start of the first is end of the second' => [
+                200000, 300000,
+                100000, 200000,
+                false,
+            ],
+            'intersection' => [
+                100000, 200000,
+                150000, 250000,
+                true,
+            ],
+        ];
+    }
+
+    /** @dataProvider providerGetIntersectionWith */
+    public function testGetIntersectionWith(
+        int $start1,
+        int $end1,
+        int $start2,
+        int $end2,
+        int $expectedStart,
+        int $expectedEnd
+    ): void {
+        $interval1 = new Interval(Instant::of($start1), Instant::of($end1));
+        $interval2 = new Interval(Instant::of($start2), Instant::of($end2));
+        $expected = new Interval(Instant::of($expectedStart), Instant::of($expectedEnd));
+
+        $this->assertEquals($expected, $interval1->getIntersectionWith($interval2));
```

Comment:
I'm not keen on using `assertEquals()` for objects. Suggestions:

- either `assertSame()` their `(string)` representation
- or implement `Interval::isEqualTo(Interval $that)`, and use it to `assertTrue()` here

Reply:
Yeah, until PHP implements proper equality override for objects, which would be very useful for value objects, it's indeed safer to use other means, although I thought it is okay enough here. By the way I dunno why it still doesn't happen, I've already seen a couple of RFCs on the topic, but haven't dived into the exact rejection reasons.